src/containers/ProfileDetails/components/ProviderId/components/OsuForm/index.js

```diff
@@ -0,0 +1,204 @@
+    >
+      {osuEnabled && (
+        <>
+          <Item
+            name="osuServerUri"
+            label="Server URI:"
+            rules={[
+              {
+                required: true,
+                message: 'Server uri field cannot be empty',
+              },
+            ]}
+          >
+            <Input placeholder="Enter a value for server uri" />
+          </Item>
+
+          <Card
+            title="Name:"
+            bordered={false}
+            extra={<Button onClick={() => setNameModal(true)}>Add</Button>}
+          >
+            <PasspointNameTable
+              tableData={osuDetails?.osuFriendlyName}
+              dataIndex="osuFriendlyName"
+              removeName={removeItem}
+            />
+          </Card>
+          <Item name="osuFriendlyName" noStyle>
+            <FormModal
+              title="Add Name"
+              fieldName="osuFriendlyName"
+              onSubmit={onSubmit}
+              visible={nameModal}
+              closeModal={handleCloseModal}
+            />
+          </Item>
+
+          <Card
+            title="Description:"
+            bordered={false}
+            extra={<Button onClick={() => setDescModal(true)}>Add</Button>}
+          >
+            <PasspointNameTable
+              tableData={osuDetails?.osuServiceDescription}
+              dataIndex="osuServiceDescription"
+              removeName={removeItem}
+            />
+          </Card>
+          <Item name="osuServiceDescription" noStyle>
+            <FormModal
+              title="Add Description"
+              fieldName="osuServiceDescription"
+              onSubmit={onSubmit}
+              visible={descModal}
+              closeModal={handleCloseModal}
+            />
+          </Item>
+
+          <Card
+            title="Icons:"
+            bordered={false}
+            extra={<Button onClick={() => setIconModal(true)}>Add</Button>}
+          >
+            <Table
+              dataSource={osuDetails?.osuIconList}
+              columns={osuIconCols}
+              pagination={false}
+              rowKey={record => record.imageUrl + record.iconLocale}
+            />
+          </Card>
+
+          <Item name="osuIconList" noStyle>
+            <Modal
+              onSuccess={handleAddIcon}
+              onCancel={() => handleCloseModal('osuIconList')}
+              title="Add Icon"
+              visible={iconModal}
+              content={
+                <Form {...layout} form={iconForm}>
+                  <Item
+                    name="imageUrl"
+                    label="Url:"
+                    rules={[
+                      {
+                        required: true,
+                        message: 'Url field cannot be empty',
+                      },
+                    ]}
+                  >
+                    <Input placeholder="Enter the image url" />
+                  </Item>
+                  <LocaleItem name="iconLocale" />
```

Comment:
Change the `PasspointNameTable` to be more dynamic so you can use it here. It should be a `PasspointLocaleTable`